src/Microsoft.ML.FastTree/GamModelParameters.cs

```diff
@@ -0,0 +1,911 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using Microsoft.ML;
+using Microsoft.ML.Calibrator;
+using Microsoft.ML.Command;
+using Microsoft.ML.CommandLine;
+using Microsoft.ML.Data;
+using Microsoft.ML.Internal.Calibration;
+using Microsoft.ML.Internal.Internallearn;
+using Microsoft.ML.Internal.Utilities;
+using Microsoft.ML.Model;
+using Microsoft.ML.Trainers.FastTree;
+using Microsoft.ML.Trainers.FastTree.Internal;
+using Microsoft.ML.Training;
+
+[assembly: LoadableClass(typeof(GamModelParametersBase.VisualizationCommand), typeof(GamModelParametersBase.VisualizationCommand.Arguments), typeof(SignatureCommand),
+    "GAM Vizualization Command", GamModelParametersBase.VisualizationCommand.LoadName, "gamviz", DocName = "command/GamViz.md")]
+
+namespace Microsoft.ML.Trainers.FastTree
+{
+    /// <summary>
+    /// The base class for GAM Model Parameters.
+    /// </summary>
+    public abstract class GamModelParametersBase : ModelParametersBase<float>, IValueMapper, ICalculateFeatureContribution,
+        IFeatureContributionMapper, ICanSaveInTextFormat, ICanSaveSummary, ICanSaveInIniFormat
+    {
+        private readonly double[][] _binUpperBounds;
+        private readonly double[][] _binEffects;
+        public readonly double Intercept;
+        private readonly int _numFeatures;
```

Comment:
>_numFeatures [](start = 29, length = 12)

I don't see any way for user to access this, but in same time it's used as constrain in public method.
Can we expose it with proper description?

Reply:
Done.